app/controllers/cloud_volume_controller.rb

```diff
@@ -569,6 +572,65 @@ def backup_restore_finished
     javascript_redirect :action => "show", :id => @volume.id
   end
 
+  def snapshot_new
+    assert_privileges("cloud_volume_snapshot_create")
+    @volume = find_by_id_filtered(CloudVolume, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Create Snapshot for %{model} \"%{name}\"") % {
+        :model => ui_lookup(:table => 'cloud_volume'),
+        :name  => @volume.name
+      },
+      :url  => "/cloud_volume/snapshot_new/#{@volume.id}"
+    )
+  end
+
+  def snapshot_create
+    assert_privileges("cloud_volume_snapshot_create")
+    @volume = find_by_id_filtered(CloudVolume, params[:id])
+    case params[:button]
+    when "cancel"
+      cancel_action(_("Snapshot of %{model} \"%{name}\" was cancelled by the user") % {
+        :model => ui_lookup(:table => 'cloud_volume'),
+        :name  => @volume.name
+      })
+    when "create"
+      options = {}
+      options[:name] = params[:snapshot_name] if params[:snapshot_name]
+      task_id = @volume.create_volume_snapshot_queue(session[:userid], options)
+      add_flash(_("Cloud volume snapshot creation failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.inspect}, :error) unless task_id.kind_of?(Fixnum)
+      if @flash_array
+        javascript_flash(:spinner_off => true)
```

Comment:
`add_flash()` and `javascript_flash()` should be replaced just by a single
`javascript_flash(:text =>, :severity =>, :spinner_off =>)`